packages/docusaurus-theme-classic/src/theme/NavbarItem/DocNavbarItem.tsx

```diff
@@ -19,9 +19,10 @@ export default function DocNavbarItem({
 }: Props): JSX.Element | null {
   const {activeDoc} = useActiveDocContext(docsPluginId);
   const doc = useLayoutDoc(docId, docsPluginId);
+  const pageActive = activeDoc?.path === doc?.path;
 
-  // Draft items are not displayed in the navbar.
-  if (doc === null) {
+  // Draft and unlisted items are not displayed in the navbar.
+  if (doc === null || (doc.unlisted && !pageActive)) {
     return null;
   }
```

Comment:
I also wish we could filter DocNavbarItem upper in the tree 🤪 , but ok for now 👍